### PR TITLE
Default firewall port to TCP when unspecified.

### PR DESCRIPTION
**Pull request by bprashanth**
Defaults to TCP so we don't allow UDP for no reason.

### CODE REVIEW

File: pkg/cloudprovider/providers/gce/gce.go [hunk 1]
```diff
@@ -1237,8 +1237,13 @@ func (gce *GCECloud) CreateFirewall(name, desc string, sourceRanges netsets.IPNe
 	// TODO: This completely breaks modularity in the cloudprovider but the methods
 	// shared with the TCPLoadBalancer take api.ServicePorts.
 	svcPorts := []api.ServicePort{}
+	// TODO: Currently the only consumer of this method is the GCE L7
+	// loadbalancer controller, which never needs a protocol other than TCP.
+	// We should pipe through a mapping of port:protocol and default to TCP
```
Comment (cjcullen) [inline pkg/cloudprovider/providers/gce/gce.go]: If we do this, we'll also need to change the functionality of the `firewallObject()` method. It currently takes a list of ports, but uses port[0]'s protocol for all of the ports. Can you add a TODO down there also? (~line 910)

**Comment by cjcullen** [comment]
Should be safe if HTTP LB is the only user. One request for a TODO, then I'll add the LGTM.

File: pkg/cloudprovider/providers/gce/gce.go (hunk 1 above)
Comment (bprashanth) [inline pkg/cloudprovider/providers/gce/gce.go]: Added TODO, PTAL.

**Comment by cjcullen** [comment]
Thanks. LGTM.

**Comment by bprashanth** [comment]
@k8s-bot unit test this github issue: #23361

**Comment by bprashanth** [comment]
@k8s-bot unit test this github issue: #23361

**Comment by bprashanth** [comment]
@k8s-bot test this github issue: #19997

**Comment by bprashanth** [comment]
@k8s-bot test this github issue: #21367

**Comment by bprashanth** [comment]
This has been in the queue for 10 days, so I'm bumping it to p0

**Comment by bprashanth** [comment]
@k8s-bot test this github issue: #23991

**Comment by bprashanth** [comment]
@k8s-bot test this github issue: https://github.com/kubernetes/kubernetes/issues/23952

**Pull request merged**